File: bounce/src/main/java/is/ricko/bounce/config/BounceDataSourceConfig.java
package is.ricko.bounce.config;

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
public class BounceDataSourceConfig {
  private final BounceConfig bounceConfig;

  @Bean
  public HikariDataSource dataSource() {
    final HikariConfig hikariConfig = new HikariConfig();
    hikariConfig.setJdbcUrl(bounceConfig.getJdbcUrl());
    hikariConfig.setUsername(bounceConfig.getJdbcUsername());
    hikariConfig.setPassword(bounceConfig.getJdbcPassword());
    return new HikariDataSource(hikariConfig);
  }
}

File: bounce/src/main/kotlin/is/ricko/bounce/config/BounceDataSourceConfig.kt
package `is`.ricko.bounce.config

import `is`.ricko.bounce.config.BounceConfig
import com.zaxxer.hikari.HikariConfig
import com.zaxxer.hikari.HikariDataSource
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class BounceDataSourceConfig @Autowired constructor(private val config: BounceConfig) {
    @Bean
    fun dataSource(): HikariDataSource {
        val hikariConfig = HikariConfig()
        hikariConfig.jdbcUrl = config.jdbcUrl
        hikariConfig.username = config.jdbcUsername
        hikariConfig.password = config.jdbcPassword
        return HikariDataSource(hikariConfig)
    }
}